src/scriptProfiler.cpp

```diff
@@ -1215,8 +1245,78 @@ class GameInstructionNewExpression : public game_instruction {
 
 #pragma endregion Instructions
 #endif
+
+std::filesystem::path findFilePath() {
+#if __linux__
+    char_t buffer[PATH_MAX] = { 0 };
+    ssize_t len = readlink("/proc/self/exe", buffer, sizeof(buffer)-1);
+    if (len != -1) {
+        buffer[len] = '\0';
+    }
+#else
+    char_t buffer[MAX_PATH] = { 0 };
+    DWORD len = GetModuleFileNameW(nullptr, buffer, MAX_PATH);
+    if (len != 0) {
+        buffer[len] = L'\0';
+    }
+#endif
+
+    string_t path{ std::move(buffer) };
+    auto pos = path.find_last_of(DIR_SEPARATOR);
+
+    // probably could do this better
+    if (pos == string_t::npos) {
+        return std::filesystem::path{"BADPATH"};
+    }
+    path = path.substr(0, pos + 1);
+
+    string_t dirName{ STR("config") };
+    string_t fileName{ STR("parameters.json") };
+    std::vector<string_t> possiblePaths;
+    // 15 was a random dice roll. Ideally we only need 1 possible path, but that will never happen in the real world.
+    possiblePaths.reserve(15);
+
+    for (auto const& dirEntry : std::filesystem::recursive_directory_iterator{ path }) {
+        if (dirEntry.is_directory()) {
+            string_t dirPath = dirEntry.path().native();
+            auto pos = dirPath.find(dirName);
+            if (pos == string_t::npos) { continue; }
+            possiblePaths.push_back(std::move(dirPath));
+        }
+    }
+
+    for (auto const& dirEntry : possiblePaths) {
+
+        std::filesystem::path filePath{dirEntry + DIR_SEPARATOR + fileName};
+        if (std::filesystem::exists(filePath)) {
+            return filePath;
+        }
+    }
+
+    return std::filesystem::path{"BADPATH"};
+}
+
 void scriptProfiler::preStart() {
     sqf::diag_log("Arma Script Profiler preStart");
+    //std::filesystem::path filePath{"config/paramaters.json"};
+
+    std::filesystem::path filePath = findFilePath();
+    hasParameterFile = (filePath.generic_string() != "BADPATH"sv);
+  
+
+    if (hasParameterFile) {
+        std::ifstream file(filePath);
+        try {
+            json = nlohmann::json::parse(file);
+            // TODO make this string view

```

Comment:
Make what string view? I don't see anything that can be

Reply:
yeah I remove the code below it, forget to remove the comment